File: hw/zehuaZhang/hw2checklists/techbowA2checklistsObjC/techbowA2checklistsObjC/IconPickerViewController.m
//
//  IconPickerViewController.m
//  techbowA2checklistsObjC
//
//  Created by ZhangZehua on 11/23/15.
//  Copyright © 2015 ZhangZehua. All rights reserved.
//

#import "IconPickerViewController.h"

@interface IconPickerViewController ()

@end

@implementation IconPickerViewController

#pragma mark - initialize

- (instancetype)initWithCoder:(NSCoder *)aDecoder
{
    self = [super initWithCoder:aDecoder];
    
    if (self) {
        _icons = @[@"No Icon",
                  @"Appointments",
                  @"Birthdays",
                  @"Chores",
                  @"Drinks",
                  @"Folder",
                  @"Groceries",
                  @"Inbox",
                  @"Photos",
                  @"Trips" ];
        _delegate = nil;
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return _icons.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"IconCell"];
    NSString *iconName = _icons[indexPath.row];
    
    cell.textLabel.text = iconName;
    cell.imageView.image = [UIImage imageNamed:iconName];
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    if (_delegate != nil) {
        NSString *iconName = _icons[indexPath.row];
        [_delegate iconPicker:self didPickIcon:iconName];
    }
}

@end

File: hw/zehuaZhang/hw2checklists/TechbbowA2CheckLists/TechbbowA2CheckLists/IconPickerViewController.swift
//
//  IconPickerViewController.swift
//  TechbbowA2CheckLists
//
//  Created by ZhangZehua on 11/19/15.
//  Copyright © 2015 ZhangZehua. All rights reserved.
//

import UIKit

protocol IconPickerViewControllerDelegate: class {
    func iconPicker(picker: IconPickerViewController, didPickIcon iconName: String)
}

class IconPickerViewController: UITableViewController {
    
    let icons = [
        "No Icon",
        "Appointments",
        "Birthdays",
        "Chores",
        "Drinks",
        "Folder",
        "Groceries",
        "Inbox",
        "Photos",
        "Trips"        
    ]
    
    weak var delegate: IconPickerViewControllerDelegate?
    
    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return icons.count
    }
    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("IconCell")
        let iconName = icons[indexPath.row]
        cell?.textLabel!.text = iconName
        cell?.imageView!.image = UIImage(named: iconName)
        
        return cell!
    }
    
    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        if let delegate = delegate {
            let iconName = icons[indexPath.row]
            delegate.iconPicker(self, didPickIcon: iconName)
        }
    }
}
